website/docs/reference/global-configs/warnings.md

```diff
@@ -2,129 +2,206 @@
 title: "Warnings"
 id: "warnings"
 sidebar: "Warnings"
+toc_max_heading_level: 2
+intro_text: "dbt's warnings can be converted to errors using --warn-error flag or WARN_ERROR config, with granular control through options."
 ---
 
-Enabling `WARN_ERROR`config or setting the `--warn-error` flag will convert dbt warnings into errors. Any time dbt would normally warn, it will instead raise an error. Examples include `--select` criteria that selects no resources, deprecations, configurations with no associated models, invalid test configurations, or tests and freshness checks that are configured to return warnings.
+Enabling `WARN_ERROR` config or setting the `--warn-error` flag will convert dbt warnings into errors. Any time dbt would normally warn, it will instead raise an error. Examples include `--select` criteria that selects no resources, deprecations, configurations with no associated models, invalid test configurations, or tests and freshness checks that are configured to return warnings.
 
  You would commonly use the `--warn-error` flag to promote test warnings from `severity: warn` to errors, but it actually affects all warning types, including:
 * Test warnings (for example, `LogTestResults`)
 * Jinja-level warnings (for example, `exceptions.warn`, or `JinjaLogWarning`)
 * Selection issues (for example, `NoNodesForSelectionCriteria`)
 * Adapter deprecation warnings (for example, `AdapterDeprecationWarning`)
 
-Consider using [`--warn-error-options`](#using-warn_error_options-for-targeted-warnings) for more targeted control over which warnings are treated as errors.
+Consider using [`--warn-error-options`](#use---warn-error-options-for-targeted-warnings) for more targeted control over which warnings are treated as errors.
 
 <File name='Usage'>
 
 ```text
 dbt --warn-error run
-...
 ```
 
 </File>
 
-### Using `--warn-error-options` for targeted warnings
+## Use `--warn-error-options` for targeted warnings
 
 Converting any warnings to errors may suit your needs perfectly, but there may be some warnings you just don't care about, and some you care about a lot. The `WARN_ERROR_OPTIONS` config or `--warn-error-options` flag gives you more granular control over _exactly which types of warnings_ are treated as errors. 
```

Comment:
```suggestion
In some cases, you may want to convert **all** warnings to errors. However, when you want _some_ warnings to stay as warnings and only promote or silence specific warnings you can instead use `--warn-error-options`. The `WARN_ERROR_OPTIONS` config or `--warn-error-options` flag gives you more granular control over _exactly which types of warnings_ are treated as errors. 
```

Reply:
warn error options allows you to specify exactly which warnings:
- remain as warnings
- are promoted to error
- are silenced